pkg/deploy/generator/resources_rp.go

```diff
@@ -900,6 +901,19 @@ mkdir -p /usr/lib/ssl/certs
 csplit -f /usr/lib/ssl/certs/cert- -b %03d.pem /etc/pki/tls/certs/ca-bundle.crt /^$/1 {*} >/dev/null
 c_rehash /usr/lib/ssl/certs
 
+# we leave clientId blank as long as only 1 managed identity assigned to vmss
+# if we have more than 1, we will need to populate with clientId used for off-node scanning
```

Comment:
Are there any checks in place to validate we only have 1 or will it be obvious if it fails?  We can't have this silently fail.

Reply:
If this changes is what will cause this to break: https://github.com/Azure/ARO-RP/blob/master/pkg/deploy/generator/resources_rp.go#L1022-L1024

Paths going forward: either add in client-id to avoid this pitfall, or add in check to confirm the map `userAssignedIdentities` has a length of 1

Reply:
Easy solution is to have RP deployment fail if more than one identity is assigned.  Rather we blow up in canary than not have this come online.  Alternately if we know azsecpack start will fail to start with invalid configuration we can rely on that to ensure no silent failures.  What is the behavior if we change nothing?

Reply:
If this config is "invalid", unfortunately, all processes will start while secretly silently failing, symptoms of failure will be visible from OS logs as the RP VM will get 403 denied while trying to download the Qualys agent.

Reply:
@bryanro92 I'd like to have something alert us when this happens.  I will follow up with MSFT on what is in place to catch this and if it's enough to cover a gap in these metrics.  If not we can create additional work items to address the gap.  In favor of merging as this is a discussion of what happens in a hypothetical future where today (pre merge) we do not have the necessary metrics at all.. so we're going to a better place without addressing this immediately.